packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -120,8 +120,14 @@ class IOSDevice extends Device {
           platformType: PlatformType.ios,
           ephemeral: true,
       ) {
-    _installerPath = _checkForCommand('ideviceinstaller');
-    _iproxyPath = _checkForCommand('iproxy');
+    if (!platform.isMacOS) {
+      printError('Cannot control iOS devices or simulators. ideviceinstaller and iproxy are not available on your platform.');
+      _installerPath = null;
+      _iproxyPath = null;
```

Comment:
return here, then else not needed.

Reply:
I chatted with @jonahwilliams about this, we it doesn't look like it gets instantiated, but we're not sure. Since it could potentially be a breaking change, I created [this issue for it](https://github.com/flutter/flutter/issues/36008) and will land after this PR.